### PR TITLE
Change stream to take in quantiles in ResetSetData

### PR DESCRIPTION
**Pull request by cw9**
Which will be used in the custom aggregation change

@xichen2020 @jeromefroe

### CODE REVIEW

File: aggregator/options.go [hunk 1]
```diff
@@ -494,6 +514,13 @@ func (o *options) FullGaugePrefix() []byte {
 	return o.fullGaugePrefix
 }
 
+func (o *options) SetTimerQuantiles(quantiles []float64) Options {
+	opts := *o
+	opts.timerQuantiles = quantiles
+	o.computeTimerQuantilesAndSuffixes()
```
Comment (xichen2020) [inline aggregator/options.go]: Should this be `opts.computeTimerQuantilesAndSuffixes()`?

File: aggregator/options.go [hunk 2]
```diff
@@ -542,7 +569,6 @@ func (o *options) computeFullPrefixes() {
 }
 
 func (o *options) computeTimerQuantilesAndSuffixes() {
```
Comment (xichen2020) [inline aggregator/options.go]: Should this be called `computeTimerQuantileSuffixes` then?

File: aggregator/types.go [hunk 3]
```diff
@@ -336,6 +339,9 @@ type Options interface {
 	// FullGaugePrefix returns the full prefix for gauges
 	FullGaugePrefix() []byte
 
+	// SetTimerQuantiles sets the timer quantiles
+	SetTimerQuantiles(quantiles []float64) Options
```
Comment (xichen2020) [inline aggregator/types.go]: These should probably be moved up to the "Read-write" base options part?

File: services/m3aggregator/config/aggregator.go [hunk 4]
```diff
@@ -176,6 +179,7 @@ func (c *AggregatorConfiguration) NewAggregatorOptions(
 	}
 	opts = opts.SetTimerQuantileSuffixFn(quantileSuffixFn)
 
+	opts = opts.SetTimerQuantiles(c.TimerQuantiles)
```
Comment (xichen2020) [inline services/m3aggregator/config/aggregator.go]: nit: can chain this up with setting the suffix fn

File: aggregator/options.go (hunk 1 above)
Comment (cw9) [inline aggregator/options.go]: good catch

File: aggregator/options.go (hunk 2 above)
Comment (cw9) [inline aggregator/options.go]: sg

File: aggregator/types.go (hunk 3 above)
Comment (cw9) [inline aggregator/types.go]: sg

File: services/m3aggregator/config/aggregator.go (hunk 4 above)
Comment (cw9) [inline services/m3aggregator/config/aggregator.go]: sg

**Pull request merged**